### ISSUE TITLE
[$125] Keyboard Navigation: Settings - Preferences: The controls are focused in a confusing order

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76924]
## Action Performed:
Prerequisite:
- User is logged in

1. Using Chrome + JAWS, open the URL: https://new.expensify.com/
2. Tab to focus on the "My account" button and press Enter to activate it
3. Once the My account menu is displayed, tab to focus on the "Preferences" menu item and press Enter to activate it
4. Tab to focus on the "Language" button and press Enter to expand it
5. Use the tab key to focus on any middle of the list options (e.g., Italiano) and press Enter to select the option
6. After selecting the option, the list closed, and again activate the language button to display the list
7. Now navigate through the list using the tab key
8. Observe the focus order of all the controls in the list

Other Occurrences:
1. Wallet - Add bank account

## Expected Result:
When reopening the language selection menu, the keyboard focus should move directly to the back button and then currently selected language option.
Users should be able to navigate the list items logically and predictably without unexpected jumps or skipped elements.

## Actual Result:
After selecting a language (e.g., Italiano), when the user opens the language menu again, the focus behaves unexpected manner. For example, after choosing "Italiano," and opening the language menu again the focus jumps to "Polski," thus skipping all the language options above then move up to "Nederlands," and finally proceeds through all remaining unselected list items.
This unpredictable focus movement creates confusion for keyboard and screen reader users and makes it difficult to understand which element is currently selected or active.

HTML:
<div class="css-g5y9jx" style="margin-top: 0px; opacity: 1; height: 640px;"><div class="css-g5y9jx" style="flex-direction: column; position: absolute; width: 359.8px; left: 0px; top: 64px;"><div class="css-g5y9jx" data-component="OfflineWithFeedback" data-element="View" data-source-file="OfflineWithFeedback.tsx"><div class="css-g5y9jx"><div class="css-g5y9jx" id="719" style="opacity: 1;"><div aria-label="English" role="presentation" data-tag="pressable" data-selection-scrapper-hidden-element="true" data-inner-box-shadow-element="false" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21" id="en" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="background-color: rgb(230, 225, 218);"><div class="css-g5y9jx r-13awgt0 r-1wtj0ep r-1awozwy r-18u37iz r-b5h31w r-1ah4tor r-9ffhgg r-lrvibr r-adacv r-kzbkwu r-ttdzmv" data-testid="base-list-item-en" style="background-color: rgb(230, 225, 218);"><div class="css-g5y9jx r-13awgt0 r-1habvwh"><div dir="auto" class="css-146c3p1 r-dnmrzs r-1udbk01 r-1iln25a" style="color: rgb(0, 46, 34); font-size: 15px; text-align: left; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 700; min-height: 20px; line-height: 20px; white-space: pre; text-decoration-line: none; overflow: hidden;">English</div></div><div class="css-g5y9jx r-18u37iz r-1awozwy r-19u6a5r"><div class="css-g5y9jx"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="20" height="20"

## Area issue was found in:
Settings-Preferences

## Failed WCAG checkpoints
WCAG 1.3.2 Meaningful Sequence
https://www.w3.org/WAI/WCAG22/Understanding/meaningful-sequence

## User impact:
A meaningful sequence that matches a logical reading order helps people who rely on assistive technologies that read content aloud. The meaning evident in the sequencing of the information in the default presentation will be the same when the content is presented in spoken form.

## Suggested resolution:
Ensure that focus is programmatically set to the currently selected language when the menu is reopened. Maintain a logical tab and arrow-key navigation order that matches the visual structure of the menu. Use element.focus() method to manage focus.

## Environment(s)
Windows 11 Home - Chrome

## Screenshots/Videos
<img width="1931" height="968" alt="Image" src="https://github.com/user-attachments/assets/df931731-17b1-4b52-8eb1-c9bba220fcef" />

https://github.com/user-attachments/assets/fb10d0cb-d46f-4644-b820-8a332ed44a2a

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6999200
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022009685473865754090</li>
        <li>Upwork Job ID: 2009685473865754090</li>
        <li>Last Price Increase: 2026-01-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ShridharGoel</details>

**Comment by rushatgabhane** [comment]
the correct amount for this issue is $125

**Comment by ShridharGoel** [comment]
@annaweber830 In your proposal, can you also include why the focused options kept jumping when it started from some middle row?

@mallenexpensify Should the selected language always be shown at the top of the list? As of now, it doesn't.

**Comment by dubielzyk-expensify** [comment]
This thing again haha. I think in this list specifically we aren't moving it to the top, but @dannymcclain can correct me

**Comment by dannymcclain** [comment]
Ugh. We have too much logic built into these lists.

>I think in this list specifically we aren't moving it to the top

I think you're right, but I can't remember the underlying reason.

**Comment by ShridharGoel** [comment]
@dubielzyk-expensify @dannymcclain Just saw https://github.com/Expensify/App/issues/69184. So, based on this looks like we are planning to move the pre-selected items to the top?

**Comment by trjExpensify** [comment]
I don't recall why this select list for languages would behave differently to https://github.com/Expensify/App/issues/69184? Is it because there isn't a search input or something? My initial take is to just show the pre-selected item on top when you navigate to the page like we're planning in that issue. 🤷
